plugins/modules/monitoring/pagerduty_user.py

```diff
@@ -0,0 +1,266 @@
+version_added: '1.3.0'
+author: Zainab Alsaffar (@zanssa)
+requirements:
+    - pdpyras python module = 4.1.1
+    - PagerDuty API Access
+options:
+    access_token:
+        description:
+            - An API access token to authenticate with the PagerDuty REST API.
+        required: true
+        type: str
+    pd_user:
+        description:
+            - Name of the user in PagerDuty.
+        required: true
+        type: str
+    pd_email:
+        description:
+            - The user's email address.
+            - I(pd_email) is the unique identifier used and cannot be updated using this module.
+        required: true
+        type: str
+    pd_role:
+        description:
+            - The user's role.
+        choices: ['global admin', 'manager', 'responder', 'observer', 'stakeholder', 'limited skateholder', 'restricted access']
+        default: 'responder'
+        type: str
+    state:
+        description:
+            - State of the user.
+            - On C(present), it creates a user if the user doesn't exist.
+            - On C(absent), it removes a user if the account exists.
+        choices: ['present', 'absent']
+        default: 'present'
+        type: str
+    pd_team:
+        description:
+            - The team to which the user belongs.
+        type: list
+        elements: str
+notes:
+    - Supports C(check_mode).
+'''
+
+EXAMPLES = r'''
+- name: Create a user account on PagerDuty
+  community.general.pagerduty_user:
+    access_token: 'Your_Access_token'
+    pd_user: user_full_name
+    pd_email: user_email
+    pd_role: user_pd_role
+    pd_team: user_pd_team
+    state: present
+
+- name: Remove a user account from PagerDuty
+  community.general.pagerduty_user:
+    access_token: 'Your_Access_token'
+    pd_user: user_full_name
+    pd_email: user_email
+    state: absent
+'''
+
+RETURN = r''' # '''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+from os import path
+
+try:
+    from pdpyras import APISession
+    HAS_PD_PY = True
+except ImportError:
+    HAS_PD_PY = False
+    PD_IMPORT_ERR = traceback.format_exc()
+
+try:
+    from pdpyras import PDClientError
+    HAS_PD_CLIENT_ERR = True
+except ImportError:
+    HAS_PD_CLIENT_ERR = False
+    PD_CLIENT_ERR_IMPORT_ERR = traceback.format_exc()
+
+
+class PagerDutyUser(object):
+    def __init__(self, module, session):
+        self._module = module
+        self._apisession = session
+
+    # check if the user exists
+    def does_user_exist(self, pd_email):
+        for user in self._apisession.iter_all('users'):
+            if user['email'] == pd_email:
+                return user['id']
+
+    # create a user account on PD
+    def add_pd_user(self, pd_name, pd_email, pd_role):
+        try:
+            user = self._apisession.persist('users', 'email', {
+                "name": pd_name,
+                "email": pd_email,
+                "type": "user",
+                "role": pd_role
```

Comment:
```suggestion
                "role": pd_role,
```
unnecessary but a good practice